docs/docs/guides/computed-attributes.mdx

```diff
@@ -0,0 +1,265 @@
+    namespace: Infra
+    label: Circuit endpoint
+    display_labels:
+      - name__value
+    attributes:
+      - name: name
+        kind: Text
+        unique: true
+    relationships:
+      - name: circuit
+        peer: InfraCircuit
+        cardinality: one
+        kind: Attribute
+      - name: location
+        peer: LocationSite
+        cardinality: one
+        kind: Attribute
+```
+
+Once completed, you can load this schema into Infrahub.
+
+:::note
+
+You can specify a transform that doesn't yet exist in Infrahub, attribute will just remain empty. Once the transform is loaded, Infrahub will automatically catch up and compute the value of this attribute.
+
+:::
+
+### Create the Python transform
+
+Please refer to the [Python transform guide](/guides/python-transform) for further details.
+
+1. First we prepare a GraphQL query that returns the data we need
+
+:::warning
+
+GraphQL query used to compute attribute's value requires `ID` as parameter.
+
+```graphql
+query MyQuery($id: ID!) {
+  MyNode(ids: [$id]) {
+    # ...
+  }
+}
+```
+
+:::
+
+```graphql title="computed_circuit_description.gql"
+query CircuitDescriptionQuery($id: ID!) {
+  InfraCircuit(ids: [$id]) {
+    edges {
+      node {
+        circuit_id {
+          value
+        }
+        endpoints {
+          edges {
+            node {
+              name { value }
+              location {
+                node {
+                  name {
+                    value
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+2. Next, the Python script that implements the logic
+
+```python title="computed_circuit_description.py"
+from infrahub_sdk.transforms import InfrahubTransform
+
+
+class ComputedCircuitDescription(InfrahubTransform):
+    query = "computed_circuit_description"
+    url = "computed_circuit_description"
+
+    async def transform(self, data):
+        circuit_dict: dict = data["InfraCircuit"]["edges"][0]["node"]
+
+        detailed_endpoints: list[str] = []
+
+        for endpoint in circuit_dict["endpoints"]["edges"]:
+            detailed_endpoints.append(
+                f'{endpoint["node"]["location"]["node"]["name"]["value"]}::{endpoint["node"]["name"]["value"]}'
+            )
+
+        return f' <- {circuit_dict["circuit_id"]["value"]} -> '.join(detailed_endpoints)
+```
+
+:::note
+
+Make sure that the script returns a string!
+
+:::
+
+3. Tie everything together into `.infrahub.yml` file
+
+```yaml title=".infrahub.yml"
+---
+python_transforms:
+  - name: computed_circuit_description
+    class_name: ComputedCircuitDescription
+    file_path: computed_circuit_description.py
+
+queries:
+  - name: computed_circuit_description
+    file_path: computed_circuit_description.gql
+```
+
+At this stage you should be able to test your transform using `infrahubctl`
+
+```shell title="❯ infrahubctl transform computed_circuit_description id=here_goes_a_circuit_id"
```

Comment:
Do we want to add a step to add data into Infrahub since we're using a new schema for this and data population will be required? This will allow the outputs for the user to match what we're showing here.

I also had to create a branch and then specify the branch for the transform to run (not sure if this is just a locally significant).

```shell
❯ infrahubctl transform computed_circuit_description id=180b4900-7ea4-f4b3-3b5f-c519831a4b93
Branch: develop not found.
```

```shell
❯ infrahubctl branch create test1
Branch 'test1' created successfully (180b4918-a3c8-859a-3561-c515b7a79f40).
❯ infrahubctl transform computed_circuit_description id=180b4900-7ea4-f4b3-3b5f-c519831a4b93 --branch test1
"den1::Test1 <- ATT123 -> dfw1::Test2"
```

Reply:
Added line for data!

Huum you shouldn't need to create a branch for that 🤔 happy to discuss it later on